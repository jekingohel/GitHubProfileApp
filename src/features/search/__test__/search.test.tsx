import React from 'react';
import {render, fireEvent, waitFor} from '@testing-library/react-native';
import Screens from '../screens';

const mockData = {
  avatar_url: 'https://dummyimage.com/100',
  name: 'John Doe',
  login: 'johndoe',
  bio: 'Software Developer',
  email: 'johndoe@example.com',
  location: 'City, Country',
  company: 'ABC Corp',
  followers: 1000,
  following: 500,
};

jest.mock('../../../requests', () => ({
  GetUser: jest.fn().mockResolvedValue(mockData),
}));

test('renders search screen', () => {
  // Render the SearchBar component
  const navigationMock = {navigate: jest.fn()};
  const {toJSON} = render(<Screens navigation={navigationMock} />);

  // Assert the presence of the search input
  expect(toJSON()).toMatchSnapshot();
});

