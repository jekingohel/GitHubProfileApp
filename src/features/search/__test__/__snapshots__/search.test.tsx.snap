// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders search screen 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    scroll={false}
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "flex": 1,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomColor": "#F4F4F4",
          "borderBottomWidth": 1,
          "columnGap": 15,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 10,
        }
      }
      testID="test-view"
    >
      <TextInput
        autoCapitalize="none"
        clearButtonMode="always"
        editable={true}
        onChangeText={[Function]}
        placeholder="Search User"
        style={
          {
            "backgroundColor": "#f4f4f4",
            "borderRadius": 10,
            "borderWidth": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontSize": 15,
            "padding": 10,
          }
        }
        testID="search-input"
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="submit-button"
      >
        <Text
          style={
            {
              "color": "#3399ff",
              "fontSize": 18,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 30,
              "textAlign": "left",
            }
          }
        >
          Submit
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingHorizontal": 30,
        }
      }
      testID="test-view"
    >
      <Text
        style={
          {
            "color": "rgb(31,35,40)",
            "fontSize": 24,
            "fontStyle": "normal",
            "fontWeight": "600",
            "lineHeight": 30,
            "textAlign": "center",
          }
        }
      >
        Find GitHub Users.
      </Text>
      <Text
        style={
          {
            "color": "rgb(101,109,118)",
            "fontSize": 18,
            "fontStyle": "normal",
            "fontWeight": "300",
            "lineHeight": 24,
            "textAlign": "center",
          }
        }
      >
        Search GitHub user and discover profiles efficiently.
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
